src/Repository/DistrictRepository.php

```diff
@@ -17,22 +15,20 @@ public function __construct(RegistryInterface $registry)
     }
 
     /**
-     * Finds referent tag for district by coordinates of the point.
+     * Finds district by coordinates of the point.
      *
-     * @return ReferentTag[]
+     * @return District[]
      */
-    public function findDistrictReferentTagByCoordinates($latitude, $longitude): array
+    public function findDistrictsByCoordinates($latitude, $longitude): array
     {
-        return $this->getEntityManager()->createQueryBuilder()
-            ->from(ReferentTag::class, 'tag')
-            ->select('tag')
-            ->innerJoin(District::class, 'district', Join::WITH, 'district.referentTag = tag')
-            ->innerJoin('district.geoData', 'geoData')
+        return $this->createQueryBuilder('district')
+            ->join('district.referentTag', 'referentTag')
```

Comment:
can you add `addSelect('referentTag')` for avoiding a additionnal query

Reply:
resolved